### PR TITLE
Ring shader

### PR DESCRIPTION
**Pull request by LGhassen**
Hello,

Here is the pull request for the ring shader. Please review and let me know if I did anything stupid.

I added a Unity project for the shader in the Shaders folder, to compile just follow the instructions included.

To use the new shader, the config for the ring must have useNewShader set to true and a penumbraMultiplier float. Example from SSRSS saturn:

			Ring
			{
				angle = 0
				outerRadius = 2500
				innerRadius = 1250
				texture = RSS-Textures/SaturnRing
				color = 1,1,1,1
				lockRotation = false
				unlit = false
				useNewShader = true
				penumbraMultipler = 10.0
			}

I posted a compiled version with the shaders here: http://forum.kerbalspaceprogram.com/index.php?/topic/140580-122-kopernicus-release-1-december-7/&do=findComment&comment=2904554

There are features I haven't done yet and which I will add later (like rings fading into noise/particules when zooming in).

Happy holidays!

### CODE REVIEW

**Comment by ThomasKerman** [comment]
Awesome! I will take a look at it immideately

File: Kopernicus/Kopernicus/Kopernicus.csproj [hunk 1]
```diff
@@ -32,6 +32,10 @@
       <HintPath>..\..\Assembly-CSharp.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Kopernicus.Parser, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (ThomasKerman) [inline Kopernicus/Kopernicus/Kopernicus.csproj]: Please revert the changes to this file. It will mess with everyones build system, since you haven't committed Kopernicus.Parser.dll (and I would prefer not to have it in the repository apart from the copy in Distribution).

With the submodule we only mess with people who do a flat checkout, and most client recognize submodules and pull them automatically theese days.

File: Shaders/KopernicusShaders/Assets/Shaders/ExportAssetBundle.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using UnityEditor;
+
+public class CreateAssetBundles
+{
+	[MenuItem ("Assets/Build AssetBundles")]
+	static void BuildAllAssetBundles ()
+	{
+		// Put the bundles in a folder called "AssetBundles"
+		var outDir = "D:/gh/Steam/steamapps/common/Kerbal Space Program/GameData/Kopernicus/Shaders";
```
Comment (ThomasKerman) [inline Shaders/KopernicusShaders/Assets/Shaders/ExportAssetBundle.cs]: Maybe use a relative path? Not blocking - I can do that myself, I just dont like hardcoded paths :D

File: Kopernicus/Kopernicus.Components/ShaderLoader.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                else if (Application.platform == RuntimePlatform.LinuxPlayer)
+                    path = path + "/Shaders/kopernicusshaders-linux";
+                else
+                    path = path + "/Shaders/kopernicusshaders-macosx";
+
+                Debug.Log("[Kopernicus] shader loader path " + path);
+
+                shaderDictionary = new Dictionary<string, Shader>();
+
+                using (WWW www = new WWW("file://" + path))
```
Comment (LGhassen) [inline Kopernicus/Kopernicus.Components/ShaderLoader.cs]: I tried asset bundle and it didn't work so I just used what works :)

**Comment by LGhassen** [comment]
Cool, I will fix the .csproj and the path asap

File: Kopernicus/Kopernicus/Kopernicus.csproj [hunk 4]
```diff
@@ -173,10 +170,6 @@
     <Compile Include="Utility.cs" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\external\config-parser\src\Kopernicus.Parser.csproj">
```
Comment (ThomasKerman) [inline Kopernicus/Kopernicus/Kopernicus.csproj]: I am not sure if deleting the reference completely is what we want :D

Comment (LGhassen) [inline Kopernicus/Kopernicus/Kopernicus.csproj]: This is why I souldn't touch code when I'm tired

Comment (LGhassen) [inline Kopernicus/Kopernicus/Kopernicus.csproj]: Should be fixed now

**Comment by ThomasKerman** [comment]
Looks good for me. Merging!
